pkg/devfile/parser/parse.go

```diff
@@ -90,83 +102,66 @@ func ParseFromData(data []byte) (d DevfileObj, err error) {
-	err = parentData.OverrideComponents(d.Data.GetParent().Components)
-	if err != nil {
-		return err
-	}
+			parentData, err := ParseFromURL(parent.Uri)
+			if err != nil {
+				return err
+			}
 
-	err = parentData.OverrideCommands(d.Data.GetParent().Commands)
-	if err != nil {
-		return err
-	}
+			parentWorkspaceContent := parentData.Data.GetDevfileWorkspace()
+			if !reflect.DeepEqual(parent.ParentOverrides, v1.ParentOverrides{}) {
+				flattenedParent, err = apiOverride.OverrideDevWorkspaceTemplateSpec(parentWorkspaceContent, parent.ParentOverrides)
+				if err != nil {
+					return err
+				}
+			} else {
+				flattenedParent = parentWorkspaceContent
+			}
 
-	err = parentData.OverrideProjects(d.Data.GetParent().Projects)
-	if err != nil {
-		return err
+			klog.V(4).Infof("adding data of devfile with URI: %v", parent.Uri)
+		}
 	}
-
-	err = parentData.OverrideStarterProjects(d.Data.GetParent().StarterProjects)
+	plugins := []*v1.DevWorkspaceTemplateSpecContent{}
+	components, err := d.Data.GetComponents(common.DevfileOptions{})
 	if err != nil {
 		return err
 	}
-
-	klog.V(4).Infof("adding data of devfile with URI: %v", parent.Uri)
-
-	// since the parent's data has been overriden
-	// add the items back to the current devfile
-	// error indicates that the item has been defined again in the current devfile
-	parentCommands, err := parentData.Data.GetCommands(common.DevfileOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error while getting commands from the parent devfiles")
-	}
-	err = d.Data.AddCommands(parentCommands...)
-	if err != nil {
-		return errors.Wrapf(err, "error while adding commands from the parent devfiles")
-	}
-
-	parentComponents, err := parentData.Data.GetComponents(common.DevfileOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error getting components from the parent devfiles")
-	}
-	err = d.Data.AddComponents(parentComponents)
-	if err != nil {
-		return errors.Wrapf(err, "error while adding components from the parent devfiles")
-	}
-
-	parentProjects, err := parentData.Data.GetProjects(common.DevfileOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error while getting projects from the parent devfiles")
-	}
-	err = d.Data.AddProjects(parentProjects)
-	if err != nil {
-		return errors.Wrapf(err, "error while adding projects from the parent devfiles")
-	}
-
-	parentStarterProjects, err := parentData.Data.GetStarterProjects(common.DevfileOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error while getting starter projects from the parent devfiles")
+	for _, component := range components {
+		if component.Plugin != nil && !reflect.DeepEqual(component.Plugin, &v1.PluginComponent{}) {
+			plugin := component.Plugin
+			var pluginData DevfileObj
+			if plugin.Uri != "" {
+				pluginData, err = ParseFromURL(plugin.Uri)
+				if err != nil {
+					return err
+				}
+			}
+			pluginWorkspaceContent := pluginData.Data.GetDevfileWorkspace()
+			result := pluginWorkspaceContent
```

Comment:
not necessary, I like your flattenedParent var, can we use it here as well? Easier to read.